### PR TITLE
New ctor for iterable as list

### PR DESCRIPTION
**Pull request by englishman**
- add new ctor into `IterableAsList`
- reuse the new ctor instead of `Arrays.asList` in other ctors

### CODE REVIEW

**Comment by 0crat** [comment]
@englishman I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/list/AllOf.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
  *
  * <pre> new AllOf(
  *   new TransformedIterable&lt;String&gt;(
- *     Arrays.asList("hello", "world"),
+ *     new IdentityArrayList<>("hello", "world"),
```
Comment (yegor256) [inline src/main/java/org/cactoos/list/AllOf.java]: @englishman typo?

**Comment by yegor256** [comment]
@englishman there seems to be a typo

File: src/main/java/org/cactoos/list/AllOf.java (hunk 1 above)
Comment (englishman) [inline src/main/java/org/cactoos/list/AllOf.java]: @yegor256 fixed

**Comment by englishman** [comment]
Can't see the latest change. Will recreate again

**Comment by ashtonhogan** [comment]
@englishman what happened with this? There seems to be a disconnect here?

**Pull request closed without merging**